src/route/routesSSRData.js

```diff
@@ -29,103 +30,120 @@ import {
-      console.log('contentLanguages', contentLanguages);
-      let file = null;
-      switch (activeTab) {
-        case 'transcription':
-        case 'research':
-        case 'articles':
-          const textFiles = transcriptGetTextFiles(unit, activeTab === 'transcription' ? '' : activeTab);
-          const transcriptLanguages = uniq(textFiles.map(x => x.language));
-          const transcriptLanguage = selectSuitableLanguage(contentLanguages, transcriptLanguages, unit.original_language, /*defaultReturnLanguage=*/ '');
-          file = transcriptSelectFile(textFiles, transcriptLanguage);
-          break;
-        case 'summary':
-          const summaryLanguages = getSummaryLanguages(unit);
-          const summaryLanguage = selectSuitableLanguage(contentLanguages, summaryLanguages, unit.original_language);
-          file = summaryGetFile(unit, summaryLanguage);
-          break;
-        default:
-          console.warn('Unsupported active tab', activeTab);
-          break;
+  if (match && match.parsedURL && match.parsedURL.searchParams) {
+    for (const [key, value] of match.parsedURL.searchParams) {
+      if (key === 'activeTab') {
+        activeTab = value;
+        break;
       }
+    }
+  }
 
-      if (file && file.id) {
-        // Load html.
-        store.dispatch(assetActions.doc2html(file.id));
-      }
+  // Select transcript file by language.
+  const contentLanguages = settingsGetContentLanguagesSelector(state);
+  let { id }             = unit;
+  let file;
+
+  switch (activeTab) {
+    case 'transcription':
+      store.dispatch(textPageActions.setFileFilter(f => f.insert_type === 'tamlil'));
+      await store.sagaMiddleWare.run(textPageSagas.fetchSubject, textPageActions.fetchSubject(id)).done;
+      file = textPageGetFileSelector(store.getState());
+      break;
+    case 'research':
+      id = Object.values(unit.derived_units).find(x => x.content_type === CT_RESEARCH_MATERIAL)?.id;
+      await store.sagaMiddleWare.run(textPageSagas.fetchSubject, textPageActions.fetchSubject(id)).done;
+      file = textPageGetFileSelector(store.getState());
+      break;
+    case 'articles':
+      id = Object.values(unit.derived_units).find(x => x.content_type === CT_ARTICLE)?.id;
+      await store.sagaMiddleWare.run(textPageSagas.fetchSubject, textPageActions.fetchSubject(id)).done;
+      file = textPageGetFileSelector(store.getState());
+      break;
+    case 'summary':
+      const summaryLanguages = getSummaryLanguages(unit);
+      const summaryLanguage  = selectSuitableLanguage(contentLanguages, summaryLanguages, unit.original_language);
+      file                   = summaryGetFile(unit, summaryLanguage);
+      break;
+    default:
+      console.warn('Unsupported active tab', activeTab);
+      break;
+  }
 
-      const c = canonicalCollection(unit);
-      if (c) {
-        store.dispatch(mdbActions.fetchCollection(c.id));
-      }
-    });
+  if (file && file.id) {
```

Comment:
@bbfsdev pls pay attention to SSR changes

Reply:
Checking...

Reply:
The SSR part works the same as in master. LGTM.